pkg/json.go

```diff
@@ -168,6 +181,74 @@ func (r *ScorecardResult) AsJSON2(showDetails bool,
 	if err := encoder.Encode(out); err != nil {
 		return sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("encoder.Encode: %v", err))
 	}
+	return nil
+}
 
+// DependencydiffResultsAsJSON exports dependencydiff results as JSON.
+func DependencydiffResultsAsJSON(depdiffResults []DependencyCheckResult,
```

Comment:
can we make this a "class" function, like we do above for `ScorecardResult`?

Reply:
> can we make this a "class" function, like we do above for `ScorecardResult`?

my initial idea is to make this func an oop-like one, but for the original scorecard run, we only return a single `ScorecardRusult`; while for this one, we return an array of `DependncydiffResult`s. 

then i just defined it this way since the `AsJSON` method does not seem to adapt to a single `DependncydiffResult`, and when we want the dependency checking results, we want an array of them (the CLI, the Action, etc.). Thus, IMO, this func should not be a class func and simply takes an array of  `DependncydiffResult` as the input.

Reply:
> > can we make this a "class" function, like we do above for `ScorecardResult`?
> 
> my initial idea is to make this func an oop-like one, but for the original scorecard run, we only return a single `ScorecardRusult`; while for this one, we return an array of `DependncydiffResult`s.
> 
> then i just defined it this way since the `AsJSON` method does not seem to adapt to a single `DependncydiffResult`, and when we want the dependency checking results, we want an array of them (the CLI, the Action, etc.). Thus, IMO, this func should not be a class func and simply takes an array of `DependncydiffResult` as the input.

oh finally this comment is not pending anymore :/